include/boost/json/monotonic_resource.hpp

```diff
@@ -20,23 +21,79 @@ namespace json {
 */
 class monotonic_resource final
     : public memory_resource
-{
+{   
+    struct initial_block
+    {
+        std::size_t size;
+        void* base = nullptr;
+        std::uintptr_t top;
+
+        initial_block() { }
```

Comment:
why not default this?

Reply:
More characters for identical initialization semantics (albeit defaulted would make the struct trivial). I'll change it.